actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java

```diff
@@ -0,0 +1,138 @@
+
+    PowerMockito.mockStatic(ServiceFactory.class);
+    PowerMockito.mockStatic(org.sunbird.common.models.util.datasecurity.impl.ServiceFactory.class);
+    EncryptionService encryptionService = Mockito.mock(EncryptionService.class);
+    Mockito.when(
+            org.sunbird.common.models.util.datasecurity.impl.ServiceFactory
+                .getEncryptionServiceInstance(null))
+        .thenReturn(encryptionService);
+
+    try {
+      Mockito.when(encryptionService.encryptData(Mockito.anyString())).thenReturn("encrptUserId");
+    } catch (Exception e) {
+      fail("AddressManagementActorTest initialization failed");
+    }
+    when(ServiceFactory.getInstance()).thenReturn(cassandraOperation);
+    when(cassandraOperation.insertRecord(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyMap()))
+        .thenReturn(getSuccessResponse());
+    when(cassandraOperation.deleteRecord(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(getSuccessResponse());
+  }
+
+  @Test
+  public void testInsertUserAddressSuccess() {
+    boolean result = testScenario(UserActorOperations.INSERT_USER_ADDRESS.getValue(), true);
+    assertTrue(result);
+  }
+
+  @Test
+  public void testUpdateAddressSuccess() {
+    boolean result = testScenario(UserActorOperations.UPDATE_USER_ADDRESS.getValue(), true);
+    assertTrue(result);
+  }
+
+  @Test
+  public void testInsertUserAddressFailureWithoutReqParams() {
+    boolean result = testScenario(UserActorOperations.INSERT_USER_ADDRESS.getValue(), false);
+    assertTrue(result);
+  }
+
+  @Test
+  public void testUpdateUserAddressFailureWithoutMandatoryFields() {
+    boolean result = testScenario(UserActorOperations.UPDATE_USER_ADDRESS.getValue(), false);
+    assertTrue(result);
+  }
+
+  private boolean testScenario(String actorOperation, boolean success) {
+    TestKit probe = new TestKit(system);
+    ActorRef subject = system.actorOf(props);
+    subject.tell(getRequestObject(actorOperation, success), probe.getRef());
+    Response res = probe.expectMsgClass(duration("10 second"), Response.class);
+    if (success) {
+      return res != null && "SUCCESS".equals(res.getResult().get(JsonKey.RESPONSE));
+    } else {
+      return res != null && res.getResult().get(JsonKey.ERROR_MSG) != null;
+    }
+  }
+
+  private Request getRequestObject(String operation, boolean withReqParams) {
+    Request reqObj = new Request();
+    reqObj.setOperation(operation);
+    if (withReqParams) {
+      reqObj.put(JsonKey.ADDRESS, getAddressList());
+      reqObj.put(JsonKey.ID, "someId");
+      reqObj.put(JsonKey.CREATED_BY, "createdBy");
+    }
+    return reqObj;
+  }
+
+  private List<Map<String, Object>> getAddressList() {
+
+    List<Map<String, Object>> lst = new ArrayList<>();
+    Map<String, Object> map = new HashMap<>();
+    map.put(JsonKey.ADDRESS, "anyAddress");
+    map.put(JsonKey.ID, "someUserId");
+    map.put(JsonKey.IS_DELETED, true);
```

Comment:
If is_deleted is false - other lines will be covered - which is for additional address addition or existing address updation.